crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2990,9 +2994,14 @@ impl<'db> TypeInferenceBuilder<'db> {
             arguments,
         } = call_expression;
 
+        let function_type = self.infer_expression(func);
+
+        if let Some(result_type) = self.try_infer_type_api_predicate(function_type, arguments) {
+            return result_type;
+        }
```

Comment:
As mentioned in another comment, I would prefer not to have special handling for "red knot type api" functions; I think they should work in the same way as `reveal_type` and `assert_type` (which are not red-knot specific).

Perhaps we don't like the way `reveal_type` is currently handled in `Type::call` and `CallOutcome`, and want to adjust it. But I don't think there should be different paths for it vs red-knot type API.